### PR TITLE
Feature/update filter api

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

Once a dataset has been filtered and uploaded to S3, post a message to the filter API with the file URL, file size, and updated status.

I have also made a few fixes as part of integration testing. 
 - The neo4j query needed some tweaks
 - exported CSV rows needed newlines between them

### How to review

Describe the steps required to test the changes.

### Who can review

Anyone

### CODE REVIEW

File: filter/store.go [hunk 1]
```diff
@@ -41,11 +43,28 @@ type HTTPClient interface {
...
+func (store *Store) PutCSVData(filterJobID string, fileURL string, size int64) error {
 
-	// todo: make call to the filter API.
+	url := store.filterAPIURL + "/filters/" + filterJobID
+
+	putBody := &observation.Filter{
+		State: "completed",
+		Downloads: &observation.Downloads{
+			CSV: &observation.DownloadItem{
+				Size: strconv.FormatInt(size, 10),
```
Comment (Matt-Guest) [inline filter/store.go:57]: Why do you need to do `strconv.FormatInt(size, 10)`?

Comment (CarlHembrough) [reply filter/store.go:57]: The size field is an int64. The typical 'strconv.itoa' function takes only the int type, not int64

File: filter/store.go [hunk 2]
```diff
@@ -41,11 +43,28 @@ type HTTPClient interface {
 }
 
 // PutCSVData allows the filtered file data to be sent back to the filter store when complete.
-func (store *Store) PutCSVData(filterJobID string, url string, size int64) error {
+func (store *Store) PutCSVData(filterJobID string, fileURL string, size int64) error {
 
-	// todo: make call to the filter API.
+	url := store.filterAPIURL + "/filters/" + filterJobID
+
+	putBody := &observation.Filter{
```
Comment (eldeal) [inline filter/store.go]: Not sure the observation package is correctly named? A filter contains many observations not the other way around? O.o Haven't looked into what all the observation package is doing to know a better name though

Comment (CarlHembrough) [reply filter/store.go]: My take on it was that its an observation filter, i.e. a filter applied to observations. I am happy to rename though if it does not make sense. I have been trying to name the packages to group related application concepts - hence observation.reader, observation.store, and observation.filter.

File: filter/store.go [hunk 3]
```diff
@@ -41,16 +44,32 @@ type HTTPClient interface {
 }
 
 // PutCSVData allows the filtered file data to be sent back to the filter store when complete.
-func (store *Store) PutCSVData(filterJobID string, url string, size int64) error {
+func (store *Store) PutCSVData(filterJobID string, fileURL string, size int64) error {
 
-	// todo: make call to the filter API.
+	url := store.filterAPIURL + "/filters/" + filterJobID
```
Comment (nshumoogum) [inline filter/store.go]: Request to filter API will need to send an `internal_token` header equal to the secret key, defaulted to `FD0108EA-825D-411C-9B1D-41EF7727F465`. Other than that the request url and body look fine to me.

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: filter/store.go (hunk 3 above)
Comment (nshumoogum) [reply filter/store.go]: The secret key should be a configurable variable.... 👍

Comment (CarlHembrough) [reply filter/store.go]: done 👍

File: filter/store.go [hunk 4]
```diff
@@ -154,6 +156,7 @@ func (store *Store) makeRequest(method, url string, body io.Reader) ([]byte, err
 	if err != nil {
 		return nil, err
 	}
+	request.Header.Set("Internal-token", store.filterAPIAuthToken)
```
Comment (nshumoogum) [inline filter/store.go]: The filter API uses snake case `internal_token` for the header name instead of `Internal-token`.

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: filter/store.go (hunk 4 above)
Comment (CarlHembrough) [reply filter/store.go]: Good catch. We will have to iron out this inconsistency across the services. I have updated it to use the correct header key.

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**